pkg/aws/core_util.go

```diff
@@ -264,17 +265,18 @@ func terminateInstance(req *driver.DeleteMachineRequest, svc ec2iface.EC2API, ma
 
 	_, err := svc.TerminateInstances(input)
 	if err != nil {
-		// if InvalidInstanceID.NotFound error from AWS, then assume VM is terminated
-		if strings.Contains(err.Error(), ec2.UnsuccessfulInstanceCreditSpecificationErrorCodeInvalidInstanceIdNotFound) {
+		// if error code is NotFound, then assume VM is terminated
+		errcode := awserror.GetMCMErrorCodeForTerminateInstances(err)
+		if errcode == codes.NotFound {
+			klog.V(2).Infof("no backing VM for %s machine found while trying to terminate instance", req.Machine.Name)
```

Comment:
update the comment to also mention , the possiblity of eventual consistency , so VM would be left, but later cleaned up by orphan collector.